common.h

```diff
@@ -35,16 +35,18 @@ typedef struct _iso_data
 
 typedef struct _choices
 {
-    int len;
-    int cur;
-    iso_data_t **values;
+    int num_allocated; /* number of items allocated in the values array */
```

Comment:
I'd suggest `capacity` instead of `num_allocated`.

Reply:
Fixed, thanks.